guides/09_other-tutorials/wrapping-layouts.md

```diff
@@ -0,0 +1,179 @@
+
+```python
+    # other LayoutBase implementations
+
+    def attach_event(self, block_dict: Dict[str, Block]) -> None:
+        raise NotImplementedError
+```
+
+You can see what is the ```block_dict``` variable in the ```Application``` class's ```attach_event``` function.
+
+### Application Class
+
+1. Render Function
+
+```python
+    # other Application implementations
+
+    def _render(self):
+        with self.app:
+            for child in self.children:
+                child.render()
+
+        self.app.render()
+```
+
+From the explanation of the ```LayoutBase``` class's ```render``` function, we can understand the ```child.render``` part.
+
+So let's look at the below part, why we are calling the ```app``` variable's ```render``` function? We are calling this render function because if we look at the implementation of this function in the ```gradio.blocks.Blocks``` class, we can see that it is adding the components and event functions into the root component. With another saying, it is creating and structuring the gradio application. So it is important to call this function.
+
+2. Attach Event Function
+
+    Let's see how we can attach events to components:
+
+```python
+    # other Application implementations
+
+    def _attach_event(self):
+        block_dict: Dict[str, Block] = {}
+
+        for child in self.children:
+            block_dict.update(child.global_children_dict)
+
+        with self.app:
+            for child in self.children:
+                try:
+                    child.attach_event(block_dict=block_dict)
+                except NotImplementedError:
+                    print(f"{child.name}'s attach_event is not implemented")
+```
+
+You can see why the ```global_children_list``` is used in the ```LayoutBase``` class from the example code. With this, all the components in the application are gathered into one dictionary, so the component can access all the components with their names.
+
+```with``` syntax is used here again to attach events to components. If we look at the ```__exit__``` function in the ```gradio.blocks.Blocks``` class, we can see that it is calling the ```attach_load_events``` function which is used for setting event triggers to components. So we have to use the ```with``` syntax to trigger the ```__exit__``` function.
+
+Yes, we can call ```attach_load_events``` function without using ```with``` syntax, but the function needs a ```Context.root_block```, and it is set in the ```__enter__``` function. So we used ```with``` syntax here rather than calling functions ourselves.
+
+## Conclusion
+
+In this guide, we saw
+
+- How we can wrap the layouts
+- How components are rendered
+- How we can structure our application with wrapped layout classes
+
+Because the classes used in this guide are used for demonstration purposes, they may lack optimizations or useful functionality. But this is the subject of another guide.
```

Comment:
```suggestion
Because the classes used in this guide are used for demonstration purposes, they may still not be totally optimized or modular. But that would make the guide much longer! 
```